src/main.rs

```diff
@@ -160,24 +179,62 @@ async fn write_state(infile: &str, outfile: &str, cachefile: &str) {
     geo_cache.save().await.expect("could not save geoip cache");
 
     let joutput = serde_json::to_string(&state).unwrap();
-    fs::write(outfile, joutput).unwrap();
+    fs::write(config.state_file_path.as_ref().unwrap(), joutput).unwrap();
 }
 
 #[tokio::main]
 async fn main() {
-    let args: Vec<String> = env::args().collect();
+    let arg_conf = ArgConfiguration::parse();
+    let mut configuration: CrunchyConfiguration;
 
-    // TODO(asmie): could be refactored to use eg. clap crate.
-    if args.len() != 4 {
-        println!("\n\nUsage is: cargo run <in-sample.json> <out-state.json> <geoip-cache.json> ");
-        println!("E.g.:     \x1b[93mcargo run --release testdata/sample.json testdata/state.json\x1b[0m\n");
-        return;
+    if let Some(config_file_path) = arg_conf.config_file {
```

Comment:
This `else.... ::default()` looks like code smell. Can `arg_conf.config_file` be mapped with `unwrap_or_default` at the end?

Reply:
Changed it - that took me a while to think it out.
Thanks!